src/config.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{assignment_logger::NoopAssignmentLogger, AssignmentLogger, EppoClient};
+
+/// Configuration for [`EppoClient`].
+pub struct Config<'a> {
+    pub(crate) api_key: String,
+    pub(crate) base_url: String,
+    pub(crate) assignment_logger: Box<dyn AssignmentLogger + Send + Sync + 'a>,
+}
+
+impl<'a> Config<'a> {
+    /// Create a default Eppo configuration using the specified API key.
+    ///
+    /// ```
+    /// # use eppo::Config;
+    /// Config::from_api_key("api-key");
+    /// ```
+    pub fn from_api_key(api_key: impl Into<String>) -> Self {
+        Config {
+            api_key: api_key.into(),
+            base_url: Config::DEFAULT_BASE_URL.to_owned(),
+            assignment_logger: Box::new(NoopAssignmentLogger),
+        }
+    }
+
+    /// Set assignment logger to pass variation assignments to your data warehouse.
+    ///
+    /// ```
+    /// # use eppo::Config;
+    /// let config = Config::from_api_key("api-key").assignment_logger(|event| {
+    ///   println!("{:?}", event);
+    /// });
+    /// ```
+    pub fn assignment_logger(
+        &mut self,
+        assignment_logger: impl AssignmentLogger + Send + Sync + 'a,
+    ) -> &mut Self {
+        self.assignment_logger = Box::new(assignment_logger);
+        self
+    }
+
+    /// Default base URL for API calls.
+    pub const DEFAULT_BASE_URL: &str = "https://fscdn.eppo.cloud/api";
```

Comment:
```suggestion
    pub const DEFAULT_BASE_URL: &'static str = "https://fscdn.eppo.cloud/api";
```

Reply:
Any reason you prefer this more verbose version?

Reply:
The Rust compiler showed this warning and even suggested this fix when I ran:

```
cargo fix --lib -p eppo
```

Reply:
Which version of rust/cargo do you use?

Reply:
Nevermind. Found it

Reply:
Just installed it yesterday!

```
cargo --version
cargo 1.77.2
```